lib/std/crypto/benchmark.zig

```diff
@@ -299,6 +299,31 @@ pub fn benchmarkAes8(comptime Aes: anytype, comptime count: comptime_int) !u64 {
     return throughput;
 }
 
+const pwhash_scrypt = [_]Crypto{Crypto{ .ty = crypto.pwhash.scrypt.kdf, .name = "scrypt" }};
+
+pub fn benchmarkPwhashScrypt(comptime kdf: anytype, comptime count: comptime_int) !u64 {
```

Comment:
Even though memory/cpu/cache-hard functions are a bit different, benchmarks compare different functions of the same class.
This is also a good opportunity to ensure that they have the same API.

`scrypt` as a password hashing function should be in a set shared with `bcrypt` (as a KDF, there's probably no need to benchmark it against PBKDF2, as benchmarking PBKDF2 is equivalent to testing the underlying hash function).

Reply:
Sensei, do we need bcrypt phc-encoding support?

Reply:
`bcrypt` has always used the `crypt` format, I'm not aware of any application using the PHC format with it. That being said, from an API perspective it doesn't really matter.

We currently have

``zig
strHash(password: []const u8, rounds_log: u6) Error![hash_length]u8
strVerify(h: [hash_length]u8, password: []const u8) Error!void
```

for bcrypt, and something that does the same thing with a different names and parameters for scrypt.

That can become:

```zig
strHash(h: []u8, password: []const u8, options: Options) Error![]u8
strVerify(h: []const u8, password: []const u8, options: Options) Error!void
```

With `Options` being a struct that sets both the output format (`.phc`, `.crypt`) and the function-specific parameters.

Even though the format can be inferred from the string, having an `Options` parameter in the verification function may not be a bad idea if we want to limit memory usage, the number of threads, etc.

Reply:
I have added options for `strHash`, but I don't know what options we expect to see in `strVerify`. Encoding can be got from str, params also encoded in str.

Reply:
`Options` can be empty for scrypt. But the verification function accepting options is going to be useful for other primitives such as Argon2 where we may want to control the number of threads.
Another option that can be useful is a callback function to monitor progress or to avoid blocking the thread during the entire computation. But that can be done later.